### ISSUE TITLE
Scipy 2.0 deprecations

### ISSUE THREAD

**Issue by heavelock** [obspy/obspy#2666]
Scipy deprecated few methods and will removed them in 2.0. Now they are popping in CI as warnings
```
scipy.polyfit is deprecated and will be removed in SciPy 2.0.0, use numpy.polyfit instead
scipy.polyval is deprecated and will be removed in SciPy 2.0.0, use numpy.polyval instead

```

**Comment by megies** [comment]
Do we know when scipy 2.0 will be out? Depending if sooner or later we might target 1.2.x instead.. dunno. Can always backport if needed though

**Comment by heavelock** [comment]
To be honest, I could not find an exact timeline. They are following 6 months release cycle and they just released 1.5 (thus the new depracation warning I think). I could not find when they want to release 2.0, they do not have a milestones ahead of 1.6. 

I would say that we have quite a bit of time :)
